checker/manual/annotating-libraries.tex

```diff
@@ -591,18 +591,47 @@ \subsection{Limitations\label{stub-limitations}}
 
 \item
   The stub file reader
-  does not handle nested classes.  To work around this, it permits a
+  does not handle nested class declarations.  To work around this, it permits a
   top-level class to be written with a \<\$> in its name, and applies the
   annotations to the appropriate nested class.
 
+\begin{Verbatim}
+    class Lib$Inner {
+       // methods
+    }
+\end{Verbatim}
+
 \item
-  Annotations must be written before the package name on a fully qualified
+  Annotations must be written before the package name on a fully-qualified
   types rather than directly on the type it qualifies.  However, it is usually not
-  necessary to write the fully qualified name.
+  necessary to write the fully-qualified name.
 
   \begin{Verbatim}
           void init(@Nullable java.security.SecureRandom random);
    \end{Verbatim}
+\item
+   Annotations on types that are inner classes must be written before the
+   package if the type is written as a
+   fully-qualified type, or before the inner class name if the type is
+   written as a simple type. For example,
```

Comment:
Werner says (the comment got wiped out because of a subsequent commit):
Can we mark the cases in the manual that violate Java 8 syntax, to highlight the discrepancies to source code?
Also, for each such mismatch, can we file an issue against the stub parser and reference the issue here.
(In Java 8 syntax, the example should be java.lang.ProcessBuilder.@NonNull Redirect.)